lib/cookies.js

```diff
@@ -1,77 +1,81 @@
-var valid_keys = ['domain', 'path', 'expires', 'httponly', 'secure'];
-
-// modified version of the parse function from the cookie lib by jshttp:
-// https://github.com/jshttp/cookie/blob/master/index.js
-
-function parse(str, opt) {
-
-  var obj   = {},
-      res   = {},
-      pairs = str.split(/; */), // matches no whitespace too
-      dec   = decodeURIComponent;
-
-  pairs.forEach(function(pair) {
-    var index = pair.indexOf('=');
-
-    // skip things that don't look like key=value
-    if (index < 0)
-      return;
-
-    var key = pair.substr(0, index).trim(),
-        val = pair.substr(++index, pair.length).trim();
-
-    // quoted values
-    if ('"' == val[0]) {
-      val = val.slice(1, -1);
-    }
-
-    // only assign once
-    if (undefined == obj[key]) {
-      try {
-        obj[key] = dec(val);
-      } catch (e) {
-        obj[key] = val;
-      }
-    }
-  });
-
-  // if key is part of valid_keys, set it, otherwise set it as key=[key] and value=[val]
-  for (var key in obj) {
-    if (valid_keys.indexOf(key.toLowerCase()) != -1) {
-      res[key]  = obj[key];
-    } else {
-      res.key   = key;
-      res.value = obj[key];
-    }
-  }
-
-  return res;
-};
-
-function to_hash(arr, url) {
-  var res = {};
-
-  arr.forEach(function(obj) {
-    // if (!obj.path || (url.indexOf(obj.path) !== -1)) {
-      res[obj.key] = obj.value;
-    // }
-  })
-
-  return res;
+//
+//  Simple cookie handling implementation based on the standard RFC 6265.
+//  This module just has two functionalities:
+//    - Parse a set-cookie-header as a key value object
+//    - Write a cookie-string from a key value object
+//  All cookie attributes are ignored.
+//
+
+// RegExps
+
+var COOKIE_PAIR = /^([^=\s]+)\s*=\s*("?)\s*(.*)\s*\2\s*$/;
+var EXCLUDED_CHARS = /[\x00-\x1F\x7F\x3B\x3B\s\"\,\\"%]/g;
+var TRAILING_SEMICOLON = /\x3B+$/;
+var SEP_SEMICOLON = /\s*\x3B\s*/;
+
+// Constants
+
+var KEY_INDEX = 1; // index of key from COOKIE_PAIR match
+var VALUE_INDEX = 3; // index of value from COOKIE_PAIR match
+
+// Convenience functions
+
+// Returns a copy str trimmed and without trainling semicolon.
+function cleanCookieString(str) {
+  return str.trim().replace(/\x3B+$/, '');
+}
+
+function getFirstPair(str) {
+  var index = str.indexOf('\x3B');
+  return index === -1 ? str : str.substr(0, index);
+}
+
+// Private functions
+
+// Returns a encoded copy of str based on RFC6265 S4.1.1.
+function encodeCookieComponent(str) {
+  return str.replace(EXCLUDED_CHARS, encodeURIComponent);
```

Comment:
This line fails if str is a number, you should to a `toString()` before replacing